tasks/FileUploadTask.md

```diff
@@ -9,26 +9,29 @@ This task aims to provide a fluent and easy to use mechanism for the consumer to
 
 - Accept a Microsoft.Graph.UploadSession instance to be used for the upload which is created and managed by the user and handed over to the task. The task should use the instance to start/resume the upload.
 - The task should validate the expiration time and expected byte range of the upload session on resume by querying the upload URL(thereby updating its instance of the Microsoft.Graph.UploadSession).
-- Accept a file/stream to be uploaded
-- The task should provide a mechanism of configuring the upload properties i.e. the upload slice size
-- The task should provide a mechanism of monitoring the upload status i.e the progress and failures through a callback/Delegate
-- Using the RequestContext, the feature flag for the FileUploadTask can be set for telemetry purposes.
-- In the event that a slice fails to upload, the task should retry to re-upload the slice for a configurable number of times(default of 3) before giving up and throwing an error/exception for the user to handle.
-- The task should be agnostic to the kind of upload being perfromed so as to support for various fileUpload scenarios e.g. DriveItem and FileAttachment
+- Accept a file/stream to be uploaded.
+- The task should provide a mechanism of configuring the upload properties i.e. the upload slice size.
+- The task should provide a mechanism of monitoring the upload progress through a callback/delegate.
+- The task should signal upload failures via standard exception mechanism.
+- The task should signal upload completion via by returning the status or returning a completed **Task**/**Promise**/**Future** for async APIs.
+- Using the RequestContext, the feature flag for the **FileUploadTask** can be set for telemetry purposes.
+- The task should not retry to upload failed slices as any retry should already be done by the retry-handler which the task should be using.
+- The task should be agnostic to the kind of upload being performed so as to support for various fileUpload scenarios e.g. **DriveItem** and **FileAttachment**.
```

Comment:
Good point.  'The task should be agnostic to the kind of upload being performed so as to support for various fileUpload scenarios'  this point can be linked to -> upload task should be marked as completed if the response status is a 201.  That is one way to achieve the loose coupling.

Reply:
Thanks! Anything you recommend we should change in this writing are can we resolve this comment?

Reply:
Move this point before "An upload task should be marked as completed if the response status.." point
- The task should be agnostic to the kind of upload being performed so as to support for various fileUpload scenarios e.g. **DriveItem** and **FileAttachment**.
- An example of the agnostic nature of task is how the task is marked as completed considering different response formats from each API.

Adding to this what else can be considered agnostic?

Reply:
I think we are good from my side

Reply:
@nikithauc I think I reorganized these points like you suggested. What do you think now?